### DISCUSSION TITLE
Access rolling average values within a `render` function

### DISCUSSION THREAD

**Discussion by jeremy-rifkin** [observablehq/plot#2209] [Q&A] [answered]
Hi,
I'm working with some modified code for a custom tooltip on multi-series plots that I got from https://github.com/observablehq/plot/issues/2003. I'm trying to get this to work with a plot I'm making where I'm using `Plot.windowY` to do a moving average, basically getting this to show the correct values: https://observablehq.com/d/bc5707e917b31108.

I've tried a number of approaches but I haven't found a way to get the window values. I was looking at https://github.com/observablehq/plot/discussions/2161 which is very close to what I'm trying to do but there doesn't seem to be a way to apply the approach used there to `render`. Is there a way to do this?

**Comment by Fil** [comment, marked as answer]
I've just sent you a suggestion, but I find it a bit too complicated for my taste.
https://observablehq.com/compare/bc5707e917b31108...4f5ee13c36704159

**Reply by jeremy-rifkin** [reply]
Thank you so much! Anything that works is perfect :) I was pulling my hair out over this. The complexity of this might be a point in favor of some library support for this type of tooltip.
